packages/wrangler/src/pages/deploy.tsx

```diff
@@ -337,9 +337,42 @@ export const Handler = async (args: PagesDeployArgs) => {
 		project_name: projectName,
 	});
 
-	logger.log(
-		`✨ Deployment complete! Take a peek over at ${deploymentResponse.url}`
-	);
+	let latestDeploymentStage: DeploymentStage | undefined;
+
+	// can this ever become an infinite loop?
```

Comment:
I'm wondering if we shouldn't put some max on how many times we want to keep on polling here 🤔 , to avoid ever running into some type of infinite loop situation

Reply:
yeah, agreed. Will fix!

Reply:
✅

Reply:
IMHO polling for 2 minutes with possibly no UI feedback (unless log level is set to debug) might be pushing it? I've set the cap to 16s, but could up to 32s. Anything above felt like a stretch of my patience as a user 😅. But I might be wrong in my thinking?

Reply:
👇 what I meant by having log level set to `debug`
<img width="849" alt="Screenshot 2024-05-14 at 20 19 28" src="https://github.com/cloudflare/workers-sdk/assets/4638332/7612addb-3659-4f9b-bd8e-7c6b5b0b3efa">

Reply:
Yeah 2m is quite a long time. We should be finishing pretty quick here, our internal timeout is 15 seconds so 16 sounds good.

Reply:
yeah, I can totes do that. Only thing that stopped me from doing that was some concern around not adding to much noise to stdout. But perhaps a one liner saying smth like `Waiting for deployment status` won't be such a big deal?